publisher/lock/s3_lock.go

```diff
@@ -90,15 +90,11 @@ func NewS3(c S3Config, logfn Logf) (*S3, error) {
 
 // Lock S3 has no compare-and-swap so this is no bulletproof solution, but should be good enough.
 func (l *S3) Lock() error {
-	var tries uint
-retry:
-	if (l.conf.MaxRetries - tries) > 0 {
-		tries++
+	for tries := 0; int(l.conf.MaxRetries) - tries > 0; tries++ {
 		l.logF("%s attempt %d", l.conf.Owner, tries)
 		if l.isBusyDeletingExpired() {
 			l.logF("%s failed, waiting %s", l.conf.Owner, l.conf.RetryBackoff.String())
 			time.Sleep(l.conf.RetryBackoff)
-			goto retry
 		}
```

Comment:
something to escape the loop is needed if not busy , could be a else+break or something similar

Reply:
what do you mean? the for condition will exit the loop

Reply:
if `if l.isBusyDeletingExpired() {` is not true i think it should exit the loop and continue with the execution as it was doing with the `goto` logic

Reply:
true! otherwise in case of long busy it will wait an extra time in the end before exiting